rust-src/crypto_common/src/types.rs

```diff
@@ -160,6 +161,26 @@ impl Deserial for OwnedContractName {
     }
 }
 
+impl Serial for Parameter<'_> {
+    fn serial<B: Buffer>(&self, out: &mut B) {
+        (self.0.len() as u16).serial(out);
+        crate::serial_vector_no_length(self.0, out)
+    }
+}
+
+impl Serial for OwnedParameter {
+    #[inline]
+    fn serial<B: Buffer>(&self, out: &mut B) { self.as_parameter().serial(out) }
+}
+
+impl Deserial for OwnedParameter {
+    fn deserial<R: ReadBytesExt>(source: &mut R) -> ParseResult<Self> {
+        let len: u16 = source.get()?;
+        let bytes = crate::deserial_vector_no_length(source, len.into())?;
```

Comment:
We could be more efficient here
```rust
let mut bytes = vec![0u8; len.into()]; // safe to preallocate since len fits u16.
source.read_exact(&mut bytes)?;
```